Description: Use SSL specific error type on SSL error
 Add debian/patches/tlscancelled.patch to fix issue with inability to
 cancel connections using unknown SSL certificates (Closes: #750995)
Origin: vendor
Author: Jim Scadden <jims@silentasylum.co.uk>
Bug-Debian: http://bugs.debian.org/750995
Bug: https://bugs.kde.org/show_bug.cgi?id=335994
Forwarded: https://bugs.kde.org/show_bug.cgi?id=335994
Reviewed-By: Scott Kitterman <scott@kitterman.com>
Last-Update: 2014-11-16

--- kdepim-runtime-4.14.2.orig/resources/imap/sessionpool.cpp
+++ kdepim-runtime-4.14.2/resources/imap/sessionpool.cpp
@@ -377,6 +377,11 @@ void SessionPool::onLoginDone( KJob *job
                                i18n( "Could not connect to the IMAP-server.\n%1",
                                      job->errorString() ) );
       }
+    } else if ( job->error() == KIMAP::LoginJob::ERR_SSL_FAILED ) {
+      cancelSessionCreation( login->session(),
+                             EncryptionError,
+                             i18n( "Could not connect to the IMAP-server %1.\n%2",
+                                   m_account->server(), job->errorString() ) );
     } else {
       // Connection worked, but login failed -> ask for a different password or ssl settings.
       m_pendingInitialSession = login->session();
